indexer/mock.go

```diff
@@ -9,25 +9,39 @@ import (
 var _ Service = (*Mock)(nil)
 
 // Mock implements a mock indexer Service
+//
+// Default Mock(s) are nil safe and return nil errors.
```

Comment:
Why do these all have nil checks now? I think I prefer the behavior where they'd panic the test if called accidentally.

Reply:
It allows for the default constructed mock to be useful in tests, instead of having to write the empty value returns often.